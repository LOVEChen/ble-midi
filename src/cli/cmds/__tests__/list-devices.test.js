/* eslint-env jest */
jest.mock('yargs');
jest.mock('../../../midi/list-devices');

test('builder(yargs)', () => {
  const mockYargs = require('yargs')();
  const builder = require('../list-devices').builder;
  const result = builder(mockYargs);
  expect(result).toEqual(mockYargs);
  expect(mockYargs.usage.mock).toMatchSnapshot();
});

test('handler', () => {
  const mockListDevices = require('../../../midi/list-devices').default;
  const handler = require('../list-devices').handler;
  handler();
  expect(mockListDevices.mock).toMatchSnapshot();
  expect(mockListDevices.mock.calls.length).toEqual(1);
  expect(mockListDevices.mock.calls[0].length).toEqual(0);
});
