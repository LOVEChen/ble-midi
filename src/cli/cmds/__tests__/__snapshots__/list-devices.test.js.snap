// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builder(yargs) 1`] = `
Object {
  "calls": Array [
    Array [
      "Usage: $0 list-devices",
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`handler 1`] = `
Object {
  "calls": Array [
    Array [],
  ],
  "instances": Array [
    undefined,
  ],
}
`;
